debugger/io.c

```diff
@@ -19,6 +19,20 @@
 #include "io.h"
 #include "websockets/packet_queue.h"
 
+#define BIT_COUNT 8
+
+static void digitalIoToString(char* const output, const uint8_t mask, const uint8_t x)
+{
+    for (size_t i = 0; i < BIT_COUNT; i++)
+    {
+        const uint8_t bit = 1 << i;
+        const bool valid = (mask & bit) != 0;
+        const bool one = (x & bit) != 0;
+        output[i] = valid ? (one ? '1' : '0') : 'X';
+    }
+    output[BIT_COUNT + 1] = 0;
```

Comment:
`'\0'` (IMHO it clearly shows were are the "string manipulation" things)

Reply:
also - why + 1?

Reply:
null terminated string :)

Reply:
'\0' - Ack
+1 - a classical off-by-one, introduced when I switched to defines... Fixing